backend/api/Services/DeckService.cs

```diff
@@ -0,0 +1,144 @@
+            _installationService = installationService;
+            _plantService = plantService;
+        }
+
+        public async Task<IEnumerable<Deck>> ReadAll()
+        {
+            return await GetDecks().ToListAsync();
+        }
+
+        private IQueryable<Deck> GetDecks()
+        {
+            return _context.Decks;
+        }
+
+        public async Task<Deck?> ReadById(string id)
+        {
+            return await GetDecks()
+                .FirstOrDefaultAsync(a => a.Id.Equals(id));
+        }
+
+        public async Task<IEnumerable<Deck>> ReadByInstallation(string installationCode)
+        {
+            var installation = await _installationService.ReadByName(installationCode);
+            if (installation == null)
+                return new List<Deck>();
+            return await _context.Decks.Where(a =>
+                a.Installation.Id.Equals(installation.Id)).ToListAsync();
+        }
+
+        public async Task<Deck?> ReadByName(string deckName)
+        {
+            if (deckName == null)
+                return null;
+            return await _context.Decks.Where(a =>
+                a.Name.ToLower().Equals(deckName.ToLower())
+            ).FirstOrDefaultAsync();
+        }
+
+        public async Task<Deck?> ReadByInstallationAndPlantAndName(Installation installation, Plant plant, string name)
+        {
+            return await _context.Decks.Where(a =>
+                a.Plant.Id.Equals(plant.Id) &&
+                a.Installation.Id.Equals(installation.Id) &&
+                a.Name.ToLower().Equals(name.ToLower())
+            ).Include(d => d.Plant).Include(i => i.Installation).FirstOrDefaultAsync();
+        }
+
+        public async Task<Deck> Create(CreateDeckQuery newDeckQuery)
+        {
+            var installation = await _installationService.ReadByName(newDeckQuery.InstallationCode);
+            if (installation == null)
+            {
+                throw new InstallationNotFoundException($"No installation with name {newDeckQuery.InstallationCode} could be found");
+            }
+            var plant = await _plantService.ReadByInstallationAndName(installation, newDeckQuery.PlantCode);
+            if (plant == null)
+            {
+                throw new PlantNotFoundException($"No plant with name {newDeckQuery.PlantCode} could be found");
+            }
+            var deck = await ReadByInstallationAndPlantAndName(installation, plant, newDeckQuery.Name);
+            if (deck == null)
+            {
+                deck = new Deck
+                {
+                    Name = newDeckQuery.Name,
+                    Installation = installation,
+                    Plant = plant
+                };
+                await _context.Decks.AddAsync(deck);
+                await _context.SaveChangesAsync();
+            }
+            return deck!;
+        }
+
+        public async Task<Deck> Update(Deck deck)
+        {
+            var entry = _context.Update(deck);
```

Comment:
We don't make any checks when updating the deck or the area, are we assuming context handles it if the deck passed does not exist? Or does the code never call this unless it exists already?

Reply:
The update method is never used, but I suppose for now we can assume the praxis would be to try catch this method. Alternatively we could try catch in here, but we would still end up returning a new exception. The documentation indicates that the object will be added if it does not exist, but I don't always trust the documentation to mention all the possible exceptions https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.dbcontext.update?view=efcore-7.0